internal/command/deploy/machines_deploymachinesapp.go

```diff
@@ -758,39 +759,93 @@ func (md *machineDeployment) updateUsingRollingStrategy(parentCtx context.Contex
 
 	for group, entries := range entriesByGroup {
 		entries := entries
-		startIdx += len(entries)
+
+		warmMachines := lo.Filter(entries, func(e *machineUpdateEntry, i int) bool {
```

Comment:
i think that having a test to make sure we're only updating a few warm machines at the same time. Your code does look solid, though.

Reply:
i'll add a test in a different PR, i'm not sure how to easily test this behaviour atm.